sdp/processors/langs/armenian.py

```diff
@@ -0,0 +1,89 @@
+import os
+import pandas as pd
+from sdp.processors.base_processor import BaseProcessor, BaseParallelProcessor, DataEntry
+from sdp.processors.modify_manifest.common import load_manifest
+
+
+class GetSource(BaseParallelProcessor):
+    """
+    A class for extracting source information from file paths and updating the dataset.
+
+    Args:
+    - input_field (str): The field containing the file path in the dataset.
+    - output_field (str): The field to store the extracted source information in the dataset.
+    - **kwargs: Additional keyword arguments to be passed to the base class `BaseParallelProcessor`.
+
+    Methods:
+    - process_dataset_entry(data_entry): Processes a single dataset entry, extracts source information, and updates the dataset.
+
+    Note:
+    - This class inherits from the `BaseParallelProcessor` class and extends its functionality to extract source information from file paths and update the dataset.
+    """
+    def __init__(
+        self,
+        input_field: str,
+        output_field: str,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.input_field = input_field
+        self.output_field = output_field
+
+    def process_dataset_entry(self, data_entry):
+        input_values = os.path.splitext(data_entry[self.input_field])[0].split("/")
```

Comment:
what exactly is this processor doing? it seems much more specific than 'GetSource'. I think it needs a more specific name, and maybe shouldn't be in a list of general processors

Reply:
true